charts/gxf/templates/service.yaml

```
@@ -1,20 +1,25 @@
 {{- $releaseName := .Release.Name -}}
 {{- $servicePort := 80 -}}
-{{- $targetPort := 8080 }}
-{{- if .Values.service }}
-  {{- $servicePort = .Values.service.port | default $servicePort }}
-  {{- $targetPort = .Values.service.targetPort | default $targetPort }}
-{{- else if .Values.httpdSidecar.enabled }}
-  {{- $servicePort = 443 }}
+{{- $targetPort := 8080 -}}
+{{- $type := "ClusterIP" -}}
+{{- if .Values.service -}}
+  {{- $servicePort = .Values.service.port | default $servicePort -}}
+  {{- $targetPort = .Values.service.targetPort | default $targetPort -}}
+  {{- if .Values.service.type -}}
+    {{- $type = .Values.service.type | default $type -}}
+  {{- end }}
+{{- else if .Values.httpdSidecar.enabled -}}
+  {{- $servicePort = 443 -}}
   {{- $targetPort = 443 -}}
-{{- end }}
+{{- end -}}
 apiVersion: v1
 kind: Service
 metadata:
   name: {{ $releaseName }}
   labels:
     app: {{ $releaseName }}
 spec:
+  type: {{ $type }}
```

Wouldn't the following be simpler and more straightforward?
```suggestion
  type: {{ .Values.service.type | default "ClusterIP" }}
```

---

`service` might not be defined, which results in an error (see the `if` around the logic above)